pkg/querier/tenantfederation/tenant_federation.go

```diff
@@ -0,0 +1,14 @@
+package tenantfederation
+
+import (
+	"flag"
+)
+
+type Config struct {
+	// Enabled switches on support for multi tenant query federation
+	Enabled bool `yaml:"enabled"`
+}
+
+func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
+	f.BoolVar(&cfg.Enabled, "tenant-federation.enabled", false, "If enabled on all components, queries can be federated across multiple tenants. The tenant IDs involved need to be specified separated by a `|` character in the `X-Scope-OrgID` header (experimental).")
```

Comment:
> `If enabled on all components`

All components is somewhat wide... this is only required in query-frontend and querier, right? No other components can use this at the moment. I think we should be bit more specific, to allow people configure it correctly and also not give them false impression that other components use it somehow.

Reply:
I have aligned that to the CHANGELOG, see https://github.com/cortexproject/cortex/pull/3250#discussion_r547935405